### DISCUSSION TITLE
DataTable rows disappear when using headerRow

### DISCUSSION THREAD

**Discussion by DaBe1812** [primefaces/community#2573] [PrimeFaces] [answered]
Hi,
I'm using Primefaces 14.0.3 and I have the issues with p:dataTable, when I'm using a p:headerRow and want to have editable="true".
Everything works fine, but if you edit a line it doesn't matter if I save or cancel the edit, the line always disappears. I only have to close and open it's headerRow and it reappears but this behavior can't be correct it must be some issue with the relöoad of the table after editing.

**Comment by melloware** [comment]
Please provide an executable example using the [PrimeFaces Test](https://github.com/primefaces/primefaces-test) project.  It is the only way developers can debug your problem to help.

**Reply by DaBe1812** [reply]
Hi, I've known that but I was to dumb t find the Test-Project again. You can find my code here
https://github.com/DaBe1812/primefaces-datatable-issue.git
When you start the server yu get a table with the artists as headerRows. By clicking on the change and commiting the change the line disappears and reappers by closing and opening the headerRow again.

**Reply by melloware** [reply]
I see. i am pretty sure Expandable Grouped header rows and editable rows was not designed to work together.  Not all features of the datatable are designed to be used together.

**Reply by DaBe1812** [reply]
But I could Imagine that it is only an issue with updating the table so it could be possible to make it work.

**Reply by melloware** [reply]
@Rapster what are you thoughts here?  This almost works but not sure these two things were designed with each other in mind.

**Reply by Rapster** [reply]
Some investigation is necessary to explain why these features are not compatible, but at first glance I'd say it's a bug

**Reply by DaBe1812** [reply]
Hi, anything new here?

**Reply by melloware** [reply]
@DaBe1812 nothing new, you can open a bug with your reproducer if you want

**Reply by melloware** [reply, marked as answer]
PR submitted: https://github.com/primefaces/primefaces/pull/13133
